ast/desugar/Desugar.cc

```diff
@@ -2199,8 +2199,13 @@ ExpressionPtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what)
                             cond = MK::If(loc, std::move(test), std::move(true_), std::move(cond));
                         }
                     }
-                    res =
-                        MK::If(when->loc, std::move(cond), node2TreeImpl(dctx, std::move(when->body)), std::move(res));
+
+                    auto whenBody = MK::Nil(when->loc);
```

Comment:
We should only need one allocation. This implementation sometimes allocates twice.